### PR TITLE
Fixed database error during distribuion creation that had a long base…

### LINKED ISSUE #1103: django.db.utils.DataError: value too long for type character varying(150) during distribution creation
**Version**
2.9

**To Reproduce**
curl -X POST https://`hostname`/pulp/api/v3/distributions/container/container/ -H 'Content-Type: application/json' -d '{"base_path":"default_organization-library-thisisaverybigcontentviewname-buttermilk_biscuits-openshift4-wincw_windows-machine-config-operator-bundle", "repository_version":"/pulp/api/v3/repositories/container/container/09248ad7-44c4-4761-917a-2734695f9eb6/versions/1/", "name":"openshift4-wincw_windows-machine-config-operator-bundle-9307"}' 

```
Traceback:

Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]: pulp [076c7a8b-e9c9-4a6f-a1a9-86a79675ad7e]: django.request:ERROR: Internal Server Error: /pulp/api/v3/distributions/container/container/
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]: Traceback (most recent call last):
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/pulp_container/app/serializers.py", line 41, in get_or_create
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    result = cls.Meta.model.objects.get(**natural_key)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    return getattr(self.get_queryset(), name)(*args, **kwargs)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/models/query.py", line 435, in get
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    raise self.model.DoesNotExist(
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]: pulp_container.app.models.ContainerNamespace.DoesNotExist: ContainerNamespace matching query does not exist.
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]: During handling of the above exception, another exception occurred:
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]: Traceback (most recent call last):
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/pulp_container/app/serializers.py", line 41, in get_or_create
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    result = cls.Meta.model.objects.get(**natural_key)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    return getattr(self.get_queryset(), name)(*args, **kwargs)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/models/query.py", line 435, in get
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    raise self.model.DoesNotExist(
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]: pulp_container.app.models.ContainerNamespace.DoesNotExist: ContainerNamespace matching query does not exist.
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]: During handling of the above exception, another exception occurred:
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]: Traceback (most recent call last):
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    return self.cursor.execute(sql, params)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]: psycopg2.errors.StringDataRightTruncation: value too long for type character varying(150)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]: The above exception was the direct cause of the following exception:
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]: Traceback (most recent call last):
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    response = get_response(request)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    response = wrapped_callback(request, *callback_args, **callback_kwargs)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    return view_func(*args, **kwargs)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/rest_framework/viewsets.py", line 125, in view
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    return self.dispatch(request, *args, **kwargs)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/rest_framework/views.py", line 509, in dispatch
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    response = self.handle_exception(exc)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/rest_framework/views.py", line 469, in handle_exception
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    self.raise_uncaught_exception(exc)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    raise exc
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/rest_framework/views.py", line 506, in dispatch
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    response = handler(request, *args, **kwargs)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/pulpcore/app/viewsets/base.py", line 448, in create
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    serializer.is_valid(raise_exception=True)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/rest_framework/serializers.py", line 227, in is_valid
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    self._validated_data = self.run_validation(self.initial_data)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/rest_framework/serializers.py", line 429, in run_validation
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    value = self.validate(value)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/pulp_container/app/serializers.py", line 288, in validate
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    validated_data["namespace"] = ContainerNamespaceSerializer.get_or_create(
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/pulp_container/app/serializers.py", line 50, in get_or_create
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    result = serializer.create(serializer.validated_data)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/pulpcore/app/serializers/base.py", line 184, in create
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    instance = super().create(validated_data)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/rest_framework/serializers.py", line 962, in create
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    instance = ModelClass._default_manager.create(**validated_data)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    return getattr(self.get_queryset(), name)(*args, **kwargs)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/models/query.py", line 453, in create
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    obj.save(force_insert=True, using=self.db)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib64/python3.9/contextlib.py", line 79, in inner
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    return func(*args, **kwds)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django_lifecycle/mixins.py", line 138, in save
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    self._run_hooked_methods(AFTER_CREATE, **kwargs)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django_lifecycle/mixins.py", line 255, in _run_hooked_methods
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    method(self)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django_lifecycle/decorators.py", line 77, in func
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    hooked_method(*args, **kwargs)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/pulpcore/app/models/access_policy.py", line 76, in add_perms
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    permission_class.handle_creation_hooks(self)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/pulpcore/app/access_policy.py", line 63, in handle_creation_hooks
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    function(creation_hook.get("permissions"), creation_hook.get("parameters"))
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/pulp_container/app/models.py", line 226, in create_namespace_group
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    group = Group.objects.create(
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    return getattr(self.get_queryset(), name)(*args, **kwargs)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/models/query.py", line 453, in create
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    obj.save(force_insert=True, using=self.db)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib64/python3.9/contextlib.py", line 79, in inner
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    return func(*args, **kwds)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django_lifecycle/mixins.py", line 134, in save
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    save(*args, **kwargs)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/models/base.py", line 739, in save
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    self.save_base(using=using, force_insert=force_insert,
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/models/base.py", line 776, in save_base
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    updated = self._save_table(
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/models/base.py", line 881, in _save_table
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    results = self._do_insert(cls._base_manager, using, fields, returning_fields, raw)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/models/sql/compiler.py", line 1416, in execute_sql
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    cursor.execute(sql, params)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/backends/utils.py", line 66, in execute
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    return executor(sql, params, many, context)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    return self.cursor.execute(sql, params)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/utils.py", line 90, in __exit__
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    raise dj_exc_value.with_traceback(traceback) from exc_value
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:  File "/usr/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]:    return self.cursor.execute(sql, params)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]: django.db.utils.DataError: value too long for type character varying(150)
Oct  4 14:35:29 sat-6-12-qa-rhel8 pulpcore-api[974]: pulp [076c7a8b-e9c9-4a6f-a1a9-86a79675ad7e]:  - - [04/Oct/2022:14:35:29 +0000] "POST /pulp/api/v3/distributions/container/container/ HTTP/1.1" 500 145 "-" "OpenAPI-Generator/2.10.3/ruby"

```
**Additional context**
https://bugzilla.redhat.com/show_bug.cgi?id=2127470

### PR DESCRIPTION
**Pull request by ipanova**
…_path.

Group name has 150 char limit https://docs.djangoproject.com/en/4.1/ref/contrib/auth/#django.contrib.auth.models.Group.name

closes #1103

### CODE REVIEW

File: pulp_container/app/models.py [hunk 1]
```diff
@@ -224,10 +224,10 @@ def create_namespace_group(self, permissions, parameters):
         group_type = parameters["group_type"]
         add_user_to_group = parameters["add_user_to_group"]
         group = Group.objects.create(
-            name="{}.{}.{}".format("container.namespace", group_type, self.name)
+            name="{}.{}.{}".format("container.namespace", group_type, self.pk)
```
Comment (ipanova) [inline pulp_container/app/models.py:227]: does anyone remember why we have decided to use namespace `name` and not `pk` for the namespace group creation? We use distribution pk for the distribution group creation.

**Comment by dralley** [comment]
Nice find!  We should probably document the `Group` model with this limitation 

https://github.com/pulp/pulpcore/blob/main/pulpcore/app/models/access_policy.py#L148

File: pulp_container/app/models.py (hunk 1 above)
Comment (mdellweg) [reply pulp_container/app/models.py:227]: Because it was easier for people to recognize?

**Comment by mdellweg** [comment]
> Nice find! We should probably document the `Group` model with this limitation
> 
> https://github.com/pulp/pulpcore/blob/main/pulpcore/app/models/access_policy.py#L148

We abandoned this pattern completely for quite some while now. Do you mean documenting on an old branch?

File: pulp_container/app/models.py (hunk 1 above)
Comment (ipanova) [reply pulp_container/app/models.py:227]: I thought so too, just wanted to make sure there was not any other reason

**Comment by dralley** [comment]
@ipanova When this gets merged please set the "fixed in version" on https://bugzilla.redhat.com/show_bug.cgi?id=2127470

File: CHANGES/1103.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+Fixed database error during distribuion creation that had a long base_path.
```
Comment (lubosmj) [inline CHANGES/1103.bugfix:1]:
```suggestion
Fixed a database error raised when creating a distribution with a long base_path.
```

**Review by lubosmj — APPROVED** [review]

**Pull request merged**